### PR TITLE
Add update information to the about dialog

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #11486

### CODE REVIEW

**Comment by erikjv** [comment]
I'm not sure if we should move everything from the settings dialog to the about dialog. I think we should move the update part in the settings up higher, because it's probably more used than bandwidth limits. 

For the about dialog: the text is customisable, so I added a label about the update status. We should probably also add a button or link text to redirect to the settings when there is an update available.

**Comment by TheOneRing** [comment]
Hmm the channel selection is only available in testpilotcloud so maybe we can move it to the about dialog?

**Comment by erikjv** [comment]
> Hmm the channel selection is only available in testpilotcloud so maybe we can move it to the about dialog?

We can. I think it would give theming/design problems if we put it in the "about" tab, so maybe make a new (second) tab called "Updates" and put the contents of the QGroupBox in there? Also, I introduced that label at the bottom to draw more attention to updates, should that label stay? (See screenshot below.)

Screenshot 2024-03-18 at 14 03 41

**Comment by TheOneRing** [comment]
I think only the text can be branded, so it should be possible to add the info into the main widget.

**Comment by erikjv** [comment]
In the latest commit I moved all of the update stuff from the settings dialog to the about dialog. @TheOneRing How can I test this?

**Comment by erikjv** [comment]
Screenshot 2024-04-05 at 16 51 05

I'm not sure if I can get the version number out. I think it'd be useful to show that too.

**Comment by erikjv** [comment]
@TheOneRing  The `OCUpdater` could provide the version. Should I add that in a separate PR?

File: src/gui/aboutdialog.cpp [hunk 1]
```diff
@@ -14,8 +14,14 @@
 #include "aboutdialog.h"
 #include "ui_aboutdialog.h"
 
-#include "theme.h"
+#include "common/restartmanager.h"
+#include "configfile.h"
```
Comment (TheOneRing) [inline src/gui/aboutdialog.cpp:18]: could you use the full paths please?

File: src/gui/aboutdialog.h [hunk 2]
```diff
@@ -34,6 +34,10 @@ class AboutDialog : public QDialog
 private:
     void openBrowser(const QString &s);
     void openBrowserFromUrl(const QUrl &s);
+    void setupUpdaterWidget();
+
+private slots:
```
Comment (TheOneRing) [inline src/gui/aboutdialog.h:39]:
```suggestion
private Q_SLOTS:
```

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/gui/aboutdialog.cpp (hunk 1 above)
Comment (erikjv) [reply src/gui/aboutdialog.cpp]: done

File: src/gui/aboutdialog.h (hunk 2 above)
Comment (erikjv) [reply src/gui/aboutdialog.h]: done

**Comment by erikjv** [comment]
> I'm not sure if I can get the version number out. I think it'd be useful to show that too.

So that was during testing. Meaning it passed the testing.

**Pull request merged**